tensornetwork/contractors/custom_path_solvers/nconinterface.py

```diff
@@ -0,0 +1,227 @@
+          log_adj[tnr[1],tnr[0]] += np.log10(tensors[tnr[0]].shape[ind[0]])
+   
+  return log_adj
+
+#------------------------------------------------------------------------#
+def ord_to_ncon(connects: List[List[int]],
+                orders: np.ndarray):
+  """
+  Produces a `ncon` compatible index contraction order from the sequence of 
+  pairwise contractions. 
+  Args:
+    connects: list of the tensor connections (in standard `ncon` format).
+    orders: array of dim (2,N-1) specifying the set of N-1 pairwise 
+      tensor contractions.
+  Returns:
+    np.ndarray: the contraction order (in `ncon` format).
+  """
+  
+  N = 1 + (orders.size//2)
+  orders = orders.reshape(2,N-1)
+  new_connects = [connects[i] for i in range(N)]
+  cont_order = np.zeros([0],dtype=int)
+  
+  for i in range(N): # always do partial traces first
+    uni_inds, counts = np.unique(new_connects[i],return_counts=True)
+    tr_inds = uni_inds[np.flatnonzero(counts == 2)]
+    cont_order = np.concatenate((cont_order,tr_inds))
+    temp_connect = np.array(new_connects[i])
+    for ind in tr_inds:
+      temp_connect = temp_connect[temp_connect != ind]
+    new_connects[i] = temp_connect 
+  
+  for i in range(N-1):  
+    # find common indices between tensor pair
+    cont_many, A_cont, B_cont = np.intersect1d(new_connects[orders[0,i]],
+                                               new_connects[orders[1,i]],  
+                                               return_indices=True)
+    temp_connects = np.append(np.delete(new_connects[orders[0,i]], A_cont), 
+                              np.delete(new_connects[orders[1,i]], B_cont))
+    cont_order = list(np.concatenate((cont_order,cont_many),axis=0))
+    
+    # build new set of connects
+    new_connects[orders[0,i]] = temp_connects
+    del new_connects[orders[1,i]]
+    
+  return cont_order
+
+#------------------------------------------------------------------------#
+def ncon_cost_check(tensor_list_in: List[np.ndarray], 
+                    connect_list_in: List[Union[List[int], Tuple[int]]], 
+                    cont_order: Optional[Union[List[int], str]] = None):
+  """
+  Checks the computational cost of an `ncon` contraction (without actually 
+  doing the contraction). Ignore the cost contributions from partial traces 
+  (which are always sub-leading).
+  Args:
+    tensors: list of the tensors in the network.
+    connects: length-N list of lists (or tuples) specifying the network 
+      connections. The jth entry of the ith list in connects labels the edge 
+      connected to the jth index of the ith tensor. Labels should be positive 
+      integers for internal indices and negative integers for free indices.
+    cont_order: optional argument to specify the order for contracting the 
+      positive indices. Defaults to ascending order if omitted. 
+  Returns:
+    float: the cost of the network contraction, given as log10(total_FLOPS).
+  """
+  
+  total_cost = None
```

Comment:
will `total_cost` always be an `int` or `float` upon returning of the function?
Why not set it to 0 here?

Reply:
Since the costs are represented in log10(FLOPS), then `total_cost=0` implies that `FLOPS=1`, so initializing `total_cost=0` can cause the FLOPS calculations to be off be 1 (which is not a big deal in practice, but was producing errors in my cost comparisons versus theoretical predictions). Technically, for a contraction that does not do any work (e.g. takes a single input tensor and leaves it alone) one should have `total_cost = -inf`. So I am not sure of what is best to return when `FLOPS=0`?

Reply:
we could return np.float('-inf') in that case

Reply:
okay, sounds good